### PR TITLE
Feature/Add CI

### PR DESCRIPTION
**Pull request by luke-hill**
This should now permit CI flows to run on this repo.

Basic pipeline that will build the image then run the default rake-task

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  checkout scm
+
+  stage("Build Image") {
+    docker.build("ca_testing")
+  }
+
+  docker.image("ca_testing").inside {
+    stage("Lint and Unit Test") {
+      sh "bundle exec rake"
+    }
```
Comment (mrdaniellewis) [inline Jenkinsfile:12]: Do you need to do the thing Zhelyan said in the software engineering channel?

```
withDockerRegistry(registry: [credentialsId: 'docker_hub']) {
    // code
}
```

**Review by mrdaniellewis — APPROVED** [review]
Looks good.  Not sure if you need the `withDockerRegistry`, which I think just logs you in.

File: Jenkinsfile (hunk 1 above)
Comment (luke-hill) [reply Jenkinsfile:12]: Probably haha. I'll add this!

Comment (zhelyan) [reply Jenkinsfile:12]: could you also remove the image when the tests are done?

```
stage("cleanup") { sh "docker rmi ca_testing -f || true" }
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+node("docker && awsaccess") {
+  cleanWs()
+  checkout scm
+
+  stage("Build Image") {
+    docker.build("ca_testing")
+  }
+
+  stage("Lint and Unit Test") {
+    withDockerRegistry(registry: [credentialsId: "docker_hub"]) {
```
Comment (zhelyan) [inline Jenkinsfile:10]: sorry mate, this should be around `docker.build()`. This is when stuff is pulled from docker hub

**Comment by zhelyan** [comment]
```
 bundle exec rake
bundler: failed to load command: rake (/app/vendor/ruby/2.7.0/bin/rake)
Bundler::GitError: The git source https://github.com/citizensadvice/citizens-advice-style-ruby.git is not yet checked out. Please run `bundle install` before trying to start your application

```

https://jenkins.devops.citizensadvice.org.uk/job/ca_testing/view/change-requests/job/PR-7/3/console

**Comment by luke-hill** [comment]
Rightio @zhelyan Not quite sure why that's happening as I run `bundle install` inside my Dockerfile. I'll take a look later today / tomorrow.

**Comment by luke-hill** [comment]
I've re-pushed, re-based and then made some tweaks to the Dockerfile as nokogiri was missing some stuff to compile.

Now getting a new issue where it says it can't run rake. I've ran identical commands (IN docker), on my local machine - At a loss why this isn't working.

Also spun up my docker container in interactive mode and ran that, all seems fine.... Bit confused what the issue is here?

**Comment by zhelyan** [comment]
```
$ whoami
whoami: unknown uid 1000
/home/ec2-user/workspace/ca_testing_PR-7 $ bundle exec rake
The git source https://github.com/citizensadvice/citizens-advice-style-ruby.git is not yet checked out. Please run `bundle install` before trying to start your application
```

no user with id 1000 in the container

**Comment by luke-hill** [comment]
@zhelyan Does this auto-build on each commit. Or have I broken something? It isn't building on the latest one currently.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+node("docker && awsaccess") {
+  cleanWs()
+  checkout scm
+
+  stage("Build Image") {
```
Comment (zhelyan) [inline Jenkinsfile:5]: if you don't plan to add more testing or publish the test reports you can reduce all this to 

```
node("docker && awsaccess") {
  cleanWs()
  checkout scm

  stage("Unit tests") {
     sh "docker-compose run --rm gem-tests"
  }
}
```

**Comment by luke-hill** [comment]
Re-ping @mrdaniellewis / @zhelyan - Is this GTM?

**Review by zhelyan — APPROVED** [review]

**Review by mrdaniellewis — APPROVED** [review]

**Pull request merged**